frame/elections-phragmen/src/benchmarking.rs

```diff
@@ -0,0 +1,283 @@
+	-> Result<(), &'static str>
+{
+	<Elections<T>>::vote(RawOrigin::Signed(caller).into(), votes, stake)
+		.map_err(|e| "failed to submit vote")
+}
+
+benchmarks! {
+	_ {}
+
+	// -- Signed ones
+	vote {
+		// range of candidates to vote for. Direct argument of the dispatchable.
+		let x in 1 .. (MAXIMUM_VOTE as u32);
+
+		// create a bunch of candidates.
+		let all_candidates = submit_candidates::<T>(MAXIMUM_VOTE as u32)?;
+
+		let caller = endowed_account::<T>("caller", SEED);
+		let stake = default_stake::<T>(BALANCE_FACTOR);
+
+		// vote first `x` ones.
+		let votes = all_candidates.into_iter().take(x as usize).collect();
+	}: _(RawOrigin::Signed(caller), votes, stake)
+
+	remove_voter {
+		// No complexity parameter. while we can vote for numerous candidates and then remove them
+		// (and I think each can have slightly different performance), we cannot express this via
+		// the parameters. Hence, we only check for the median, 8 votes to be removed. Only use
+		// account seed as complexity parameter.
+		// TODO: what should be the upper bound here? the first benchmark runs 16 iterations.. this
+		// one 1000? These are used only to iterate and run the benchmark s times. They are not a
+		// parameter in any way.
+		let s in 0 .. 1000;
+
+		let votes_to_remove = (MAXIMUM_VOTE / 2) as u32;
+
+		// create a bunch of candidates.
+		let all_candidates = submit_candidates::<T>(votes_to_remove)?;
+
+		let caller = endowed_account::<T>("caller", s);
+		let stake = default_stake::<T>(BALANCE_FACTOR);
+		submit_voter::<T>(caller.clone(), all_candidates, stake)?;
+	}: _(RawOrigin::Signed(caller))
+
+	report_defunct_voter {
+		// has no complexity parameter. two voters exist. One only votes fro outdated candidates.
+		// The other one can report the defunct one.
+		let s in 0 .. 1000;
+
+		// create a bunch of candidates.
+		let candidate_count = (MAXIMUM_VOTE) as u32;
+		let all_candidates = submit_candidates::<T>(candidate_count)?;
+
+		let stake = default_stake::<T>(BALANCE_FACTOR);
+
+		// account 1 votes for the first half
+		let account_1 = endowed_account::<T>("caller_1", s);
+		let bailing = all_candidates
+			.iter()
+			.take(candidate_count as usize / 2)
+			.cloned()
+			.collect::<Vec<_>>();
+
+		submit_voter::<T>(
+			account_1.clone(),
+			bailing.clone(),
+			stake,
+		)?;
+		let account_1_lookup = as_lookup::<T>(account_1);
+
+		// account 2 votes for the second half
+		let account_2 = endowed_account::<T>("caller_2", s);
+		submit_voter::<T>(
+			account_2.clone(),
+			all_candidates.iter().rev().take(candidate_count as usize / 2).cloned().collect(),
+			stake,
+		)?;
+
+		// all the first chunk of candidates bail out
+		bailing.into_iter().for_each(|b| {
+			assert_ok!(<Elections<T>>::renounce_candidacy(RawOrigin::Signed(b).into()));
+		});
+	}: _(RawOrigin::Signed(account_2), account_1_lookup)
+
+	submit_candidacy {
+		// No complexity parameter. Use account seed to iterate.
```

Comment:
`is_candidate`, `is_member`, is_runner` is all used here again, so you need to populate these vectors